fetch.bs

```diff
@@ -8485,6 +8618,92 @@ with a <var>promise</var>, <var>request</var>, <var>responseObject</var>, and an
 </div>
 
 
+<h3 id=fetch-later-method>FetchLater method</h3>
+
+<pre class=idl>
+
+dictionary DeferredRequestInit : RequestInit {
+  DOMHighResTimeStamp backgroundTimeout;
```

Comment:
Linking to https://github.com/WICG/pending-beacon/issues/13#issuecomment-1711315654 where we argue that a general `timeout` mechanism may be more ergonomic to support the user story "send this request after 10minute from now", in addition to `backgroundTimeout`.

Reply:
Not just more ergonomic. It seems impossible to use timers realiably on Chrome and FF mobile once the page loses visibility.

Reply:
If that's the case, what's the point of this timeout at all? Perhaps we should spec & ship a substantially simpler version without it, and people can implement a regular timeout in userland?

Reply:
> If that's the case, what's the point of this timeout at all? Perhaps we should spec & ship a substantially simpler version without it, and people can implement a regular timeout in userland?

The point here you can't even do a reliable timer with userland + `backgroundTimeout`. With userland-only it's even worse.

So the argument is to make this `timeout`, not `backgroundTimeout`. An uncacenlled fetch will activate on this timer whether the page is visible, invisible, frozen or in BFCache.

Reply:
A side note, AbortSignal also provides a timeout mechanism, but it is based on active time, so it's still not equivalent to "send this X min from now" behavior.

```
AbortSignal: timeout() static method

The timeout is based on active rather than elapsed time, and will effectively be paused if the code is running in a suspended worker, or while the document is in a back-forward cache.
```